core/files/assets/files-upload-handler.php

```diff
@@ -45,15 +45,15 @@ final public static function is_enabled() {
 		 *
 		 * @since 3.0.0
 		 *
-		 * @param bool $enabled Weather upload is enabled or not.
+		 * @param bool $enabled Whether upload is enabled or not.
 		 */
 		$enabled = apply_filters( 'elementor/files/allow_unfiltered_upload', $enabled );
 
 		return $enabled;
 	}
 
 	final public function support_unfiltered_files_upload( $existing_mimes ) {
-		if ( $this->is_elementor_media_upload() ) {
+		if ( $this->is_elementor_media_upload() && $this->is_enabled() ) {
```

Comment:
Isn't it handled [here](https://github.com/elementor/elementor/blob/master/core/files/assets/files-upload-handler.php#L83)?

Reply:
Looks like it doesn't 🤷‍♂️ 
I have no idea why, it LGTM.